warehouse/query-core/src/main/java/datawave/query/predicate/EventDataQueryExpressionFilter.java

```diff
@@ -56,7 +70,10 @@ public void startNewDocument(Key document) {
     
     @Override
     public boolean keep(Key k) {
-        return true;
+        DatawaveKey key = new DatawaveKey(k);
+        
+        // keep as long as it is part of the event or it is part of the query
+        return !nonEventFields.contains(key.getFieldName()) || apply(new AbstractMap.SimpleEntry<>(k, null));
```

Comment:
So apply means add to document, keep means send back to user (but only those actually added to the document).
This says we are sending back to the user anything we add to the document. Is that what we want?

Reply:
If this is actually dead code, then return true.

Reply:
next commit will have true